ui/selectors/selectors.js

```diff
@@ -403,6 +403,28 @@ export function getUnapprovedConfirmations(state) {
   return Object.values(pendingApprovals);
 }
 
+export function getFailedTransactionsToDisplay(state) {
+  const { currentNetworkTxList = [] } = state.metamask;
+  const { transactionsToDisplayOnFailure = {} } = state.appState;
+  const failedTransactions = currentNetworkTxList.reduce(
+    (_failedTransactions, tx) => {
+      if (
+        transactionsToDisplayOnFailure[tx.id] &&
+        (tx.err || tx.status === 'approved' || tx.status === 'signed')
+      ) {
```

Comment:
I don't think we need to filter transactions by state or by the presence of an error here. If a transaction is in the `transactionsToDisplayOnFailure` map, we should include it here. The logic for filtering out completed transactions is already present in the background state update handler - we don't need to duplicate it.

This is dangerous in that it could hide an error if we did include a transaction by mistake that doesn't belong here. Also it would result in a mismatch between this list and the "display count" derived below.

```suggestion
      if (transactionsToDisplayOnFailure[tx.id]) {
```